dranspose/ingester.py

```diff
@@ -298,28 +306,24 @@ async def work(self) -> None:
                 self._logger.debug("workermessages %s", workermessages)
                 await self._send_workermessages(workermessages)
                 end = time.perf_counter()
-                took.append(end - start)
-                if len(took) > 1000:
+                time_spent_per_assignment.append(end - start_time)
+                if len(time_spent_per_assignment) > 1000:
                     self._logger.info(
-                        "forwarding took avg %lf, min %f max %f",
-                        sum(took) / len(took),
-                        min(took),
-                        max(took),
+                        "forwarding took avg %lf, min %f max %f. waiting time %f%",
+                        sum(time_spent_per_assignment) / len(time_spent_per_assignment),
+                        min(time_spent_per_assignment),
+                        max(time_spent_per_assignment),
+                        time_spent_waiting / sum(time_spent_per_assignment) * 100,
                     )
-                    self._logger.info("waiting for queue %d of 1000", len(empties))
-                    took = []
-                    empties = []
+                    # reset counters
+                    time_spent_per_assignment = []
+                    time_spent_waiting = 0
                 self.state.processed_events += 1
         except asyncio.exceptions.CancelledError:
             self._logger.info("stopping worker")
             for stream in self.active_streams:
                 await self.stop_source(stream)
-            if self.dump_file:
-                self._logger.info(
-                    "closing dump file %s at cancelled work", self.dump_file
-                )
-                self.dump_file.close()
-                self.dump_file = None
+            del dumper
```

Comment:
maybe we need this explicit `del` in the `restart_work` as well.

Reply:
It now closes in `finish_work` as well as a `finally` block in `work`. That should cover it.